src/views/virtualmachines/details/tabs/snapshots/SnapshotListPage.tsx

```diff
@@ -18,17 +23,38 @@ type SnapshotListPageProps = RouteComponentProps<{
   obj?: V1VirtualMachine;
 };
 
-const SnapshotListPage: React.FC<SnapshotListPageProps> = ({ obj }) => {
+const SnapshotListPage: React.FC<SnapshotListPageProps> = ({ obj: vm }) => {
   const { t } = useKubevirtTranslation();
+  const [isOpen, setIsOpen] = React.useState(false);
+  const { snapshots, restoresMap, loaded, error } = useSnapshotData(
+    vm?.metadata?.name,
+    vm?.metadata?.namespace,
+  );
 
   return (
     <>
       <ListPageHeader title="">
-        <ListPageCreateButton>{t('Add snapshot')}</ListPageCreateButton>
+        <ListPageCreateButton onClick={() => setIsOpen(true)}>
+          {t('Add snapshot')}
+        </ListPageCreateButton>
       </ListPageHeader>
       <ListPageBody>
-        <SnapshotList vm={obj} />
+        <SnapshotList
+          snapshots={snapshots}
+          restoresMap={restoresMap}
+          loaded={loaded}
+          error={error}
+          isVMRunning={vm?.status?.printableStatus !== printableVMStatus.Stopped}
+        />
       </ListPageBody>
+      {isOpen && (
+        <SnapshotModal
+          usedNames={getUsedSnapshotNames(snapshots)}
```

Comment:
why do we need this client side logic? the backend should reject a request if a snapshot with the same name is being created

Reply:
np, will remove